From c44b9c919e9604390f9fbbfefc45d53e46000bde Mon Sep 17 00:00:00 2001
From: Kexy Biscuit <kexybiscuit@aosc.io>
Date: Tue, 8 Oct 2024 00:15:27 +0800
Subject: [PATCH 133/135] AOSCOS: Input: Fix detection logic of PixArt PS/2
 touchpad

Previous detection code of PixArt assigns PixArt clickpad to all devices
that are not PixArt touchpads. Current detection code returns -ENODEV
for them.

Cc: Huacai Chen <chenhuacai@kernel.org>
Fixes: 740ff03d7238 ("Input: Add driver for PixArt PS/2 touchpad")
Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 drivers/input/mouse/pixart_ps2.c | 23 ++++++++++++++++++++---
 1 file changed, 20 insertions(+), 3 deletions(-)

diff --git a/drivers/input/mouse/pixart_ps2.c b/drivers/input/mouse/pixart_ps2.c
index 1993fc760d7b..6f937bf42fde 100644
--- a/drivers/input/mouse/pixart_ps2.c
+++ b/drivers/input/mouse/pixart_ps2.c
@@ -69,7 +69,16 @@ static int pixart_read_tp_type(struct ps2dev *ps2dev, u8 *type)
 	if (error)
 		return error;
 
-	*type = param[0] == 0x0e ? PIXART_TYPE_TOUCHPAD : PIXART_TYPE_CLICKPAD;
+	switch (param[0]) {
+		case (0x0e):
+			*type = PIXART_TYPE_TOUCHPAD;
+			break;
+		case (0x0c):
+			*type = PIXART_TYPE_CLICKPAD;
+			break;
+		default:
+			return -ENODEV;
+	}
 
 	return 0;
 }
@@ -237,8 +246,16 @@ int pixart_detect(struct psmouse *psmouse, bool set_properties)
 
 	if (set_properties) {
 		psmouse->vendor = "PixArt";
-		psmouse->name = (type == PIXART_TYPE_TOUCHPAD) ?
-				"touchpad" : "clickpad";
+		switch (type) {
+			case (PIXART_TYPE_TOUCHPAD):
+				psmouse->name = "touchpad";
+				break;
+			case (PIXART_TYPE_CLICKPAD):
+				psmouse->name = "clickpad";
+				break;
+			default:
+				return -ENODEV;
+		}
 	}
 
 	return 0;
-- 
2.47.0

